custom/icds/management/commands/find_data_not_in_es.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import csv342 as csv
+from datetime import date, timedelta
+
+from django.core.management import BaseCommand
+
+from corehq.apps.es import CaseES, FormES
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.models import CommCareCaseSQL, XFormInstanceSQL
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+
+from io import open
+
+
+class Command(BaseCommand):
+    """https://manage.dimagi.com/default.asp?277644
+    https://trello.com/c/gLN9bxOt/389-backfill-missing-data-in-es
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'csv_file',
+            help="File path for csv file",
+        )
+
+    def handle(self, csv_file, **options):
+        self.domain = 'icds-cas'
+        self.case_accessor = CaseAccessors(self.domain)
+        with open(csv_file, "w", encoding='utf-8') as csv_file:
+            field_names = ('date', 'doc_type', 'in_sql', 'in_es',)
+
+            csv_writer = csv.DictWriter(csv_file, field_names, extrasaction='ignore')
+            csv_writer.writeheader()
+
+            current_date = date(2017, 1, 1)
+
+            while current_date <= date.today():
+                properties = {
+                    "date": current_date,
+                    "doc_type": "CommCareCase",
+                    "in_sql": self._get_sql_cases_modified_on_date(current_date),
+                    "in_es": self._get_es_cases_modified_on_date(current_date),
+                }
+                csv_writer.writerow(properties)
+
+                properties = {
+                    "date": current_date,
+                    "doc_type": "XFormInstance",
+                    "in_sql": self._get_sql_forms_modified_on_date(current_date),
+                    "in_es": self._get_es_forms_modified_on_date(current_date),
+                }
+                csv_writer.writerow(properties)
+
+                current_date += timedelta(days=1)
+
+    def _get_sql_cases_modified_on_date(self, date):
+        num_cases = 0
+        dbs = get_db_aliases_for_partitioned_query()
+        for db in dbs:
+            num_cases += (
+                CommCareCaseSQL.objects
+                .using(db)
+                .filter(domain=self.domain, modified_on__date=date)
```

Comment:
Using `modified_on` won't really give much useful info I don't think since most cases will have been modified since their original creation. Would it not be better to use `opened_on`?

Reply:
I did this intentionally. My theory was that these were mostly effecting cases that had not been updated (mostly households) so `modified_on` would give us the day we last expected it to be updated. If we were using creation time, then older versions of the cases could be in ES